gcpauth/token.go

```diff
@@ -22,6 +23,13 @@ func isEnvExist(key string) bool {
 // If the GOOGLE_APPLICATION_CREDENTIALS environment variable is set, it will read an auth.json file from the path
 // If it isn't set, it will use the use internal GCP mechanism to authenticate it's self.
 func GetAuthToken(saEmail string) (string, error) {
+	// getting a token needs this to be appended, so automatically add it if it's not there
+	serviceAcctRegex := regexp.MustCompile(`\.gserviceaccount\.com$`)
```

Comment:
Regex compilation is expensive.
Please move it out of the function.

Reply:
Auto appending `.gserviceaccount.com` make it harder to use default credentials.
We need to generate key for every service during development.

Making it harder for some people.

Reply:
What do you mean need to generate a key for every service during development?

Reply:
Assume, that I am working on a service but I cannot use ADC because the code keeps on appending `.gserviceaccount.com` so, to overcome that I have to create or use a service account and download it to use it.